File: src/main/java/com/procurement/submission/model/entity/BidEntity.java
package com.procurement.submission.model.entity;

import java.time.LocalDateTime;
import java.time.ZoneOffset;
import java.util.Date;
import java.util.UUID;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.Column;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

@Getter
@Setter
@Table("submission_bid")
public class BidEntity {
    @PrimaryKeyColumn(name = "cp_id", type = PrimaryKeyType.PARTITIONED)
    private String cpId;

    @PrimaryKeyColumn(name = "stage", type = PrimaryKeyType.CLUSTERED)
    private String stage;

    @PrimaryKeyColumn(name = "bid_id", type = PrimaryKeyType.CLUSTERED)
    private UUID bidId;

    @PrimaryKeyColumn(name = "token_entity", type = PrimaryKeyType.CLUSTERED)
    private UUID token;

    @Column("owner")
    private String owner;

    @Column(value = "status")
    private String status;

    @Column("created_date")
    private Date createdDate;

    @Column("pending_date")
    private Date pendingDate;

    @Column(value = "json_data")
    private String jsonData;

    public LocalDateTime getCreatedDate() {
        return LocalDateTime.ofInstant(createdDate.toInstant(), ZoneOffset.UTC);
    }

    public void setCreatedDate(final LocalDateTime createdDate) {
        this.createdDate = Date.from(createdDate.toInstant(ZoneOffset.UTC));
    }

    public LocalDateTime getPendingDate() {
        return LocalDateTime.ofInstant(pendingDate.toInstant(), ZoneOffset.UTC);
    }

    public void setPendingDate(final LocalDateTime pendingDate) {
        this.pendingDate = Date.from(pendingDate.toInstant(ZoneOffset.UTC));
    }
}

File: src/main/kotlin/com/procurement/submission/application/repository/bid/model/BidEntity.kt
package com.procurement.submission.application.repository.bid.model

import com.procurement.submission.domain.model.Cpid
import com.procurement.submission.domain.model.Ocid
import com.procurement.submission.domain.model.Owner
import com.procurement.submission.domain.model.Token
import com.procurement.submission.domain.model.bid.BidId
import com.procurement.submission.domain.model.enums.Status
import com.procurement.submission.model.dto.ocds.Bid
import java.time.LocalDateTime

sealed class BidEntity {
    abstract val cpid: Cpid
    abstract val ocid: Ocid
    abstract val createdDate: LocalDateTime
    abstract val pendingDate: LocalDateTime?

    class New(
        override val cpid: Cpid,
        override val ocid: Ocid,
        val token: Token,
        val owner: Owner,
        override val createdDate: LocalDateTime,
        override val pendingDate: LocalDateTime?,
        val bid: Bid
    ) : BidEntity()

    class Updated(
        override val cpid: Cpid,
        override val ocid: Ocid,
        override val createdDate: LocalDateTime,
        override val pendingDate: LocalDateTime?,
        val bid: Bid
    ) : BidEntity()

    class Record(
        val cpid: Cpid,
        val ocid: Ocid,
        val bidId: BidId,
        val token: Token,
        val owner: Owner,
        val status: Status,
        val createdDate: LocalDateTime,
        val pendingDate: LocalDateTime?,
        val jsonData: String
    )
}
